src/api/util.ts

```diff
@@ -46,7 +48,17 @@ export class Session {
     async safeFetch(input: RequestInfo, init?: RequestInit): Promise<Response> {
         const response = await this.fetch(input, init);
         if(response.status >= 400) {
-            throw new StatusError(response.status, response.statusText, response);
+            try {
+                const json = await response.json();
+
+                if(typeof json.message === "string") {
+                    throw new ApiError(json.message);
+                }
+
+                throw new StatusError(response.status, response.statusText, response);
+            } catch(e) {
+                throw new StatusError(response.status, response.statusText, response);
```

Comment:
Ehh, this is exactly the same as the thing above. So you throw an error, then catch it, and then in the catch throw the exact same error. Either move this throw (and the other) below the try, or remove one of them.